File: LongestCommonPrefix.py
"""
14. Longest Common Prefix
Write a function to find the longest common prefix string amongst an array of strings.
"""
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ''
        res = 0
        while True:
            for i in range(len(strs)):
                if res >= len(strs[i]):
                    return strs[0][0 : res]
                if strs[i][res] != strs[0][res]:
                    return strs[0][0 : res]
            res += 1

if __name__ == "__main__":
    strs = ["asc", "as", "a"]
    print(Solution().longestCommonPrefix(strs))

File: Java/LongestCommonPrefix.java

public class LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        StringBuffer res = new StringBuffer();
        if(strs.length == 0) {
            return "";
        }
        for(int i = 0; i < strs[0].length(); i++) {
            res.append(strs[0].charAt(i));
        }
        for(int i = 1; i < strs.length; i++) {
            if(strs[i].length() < res.length()) {
                res.delete(strs[i].length(), res.length());
            }
            for(int j = 0; j < res.length(); j++) {
                if(strs[i].charAt(j)!= res.charAt(j)) {
                    res.delete(j, res.length());
                }
            }
        }
        return res.toString();
    }

	public static void main(String[] args) {


	}

}
